cf/commands/target.go

```diff
@@ -54,17 +60,34 @@ func (cmd Target) GetRequirements(requirementsFactory requirements.Factory, c *c
 		return
 	}
 
-	reqs = append(reqs, requirementsFactory.NewApiEndpointRequirement())
+	if c.String("a") == "" {
+		reqs = append(reqs, requirementsFactory.NewApiEndpointRequirement())
+	}
+
 	if c.String("o") != "" || c.String("s") != "" {
 		reqs = append(reqs, requirementsFactory.NewLoginRequirement())
 	}
 
+	if !cmd.config.IsLoggedIn() && c.String("a") != "" && (c.String("o") != "" || c.String("s") != "") {
+		err = errors.NewWithFmt(T("should not use [-o] and [-s] when user is not logged in"))
```

Comment:
The error can be more positive, sth like, "user not logged in, cannot use -o or -s flag to target"